device/tt_silicon_driver.cpp

```diff
@@ -1174,19 +1232,24 @@ void set_use_dma(bool msi, uint32_t dma_block_size_read_threshold_bytes, uint32_
     g_DMA_BLOCK_SIZE_WRITE_THRESHOLD_BYTES = dma_block_size_write_threshold_bytes;
 }
 
+void write_regs(volatile uint32_t *dest, const uint32_t *src, uint32_t word_len) {
+    while (word_len-- != 0) {
+        uint32_t temp;
```

Comment:
I see that this is refactored rather than new code.  But I wonder if
```
    while (word_len-- != 0) {
        *dest++ = *src++;
    }
```
is a clearer way to express it?

Reply:
Don't want to spend much time on refactoring existing code at this point. But I will edit and test this.